src/docfx/docset/Context.cs

```diff
@@ -142,6 +146,14 @@ private sealed class Cache
                     return JsonUtility.Deserialize(content);
                 })).Value;
 
+            public (List<Error> errors, JObject metadata) ExtractMetadata(Document file)
+                => _metadataCache.GetOrAdd(GetKeyFromFile(file), new Lazy<(List<Error>, JObject)>(() =>
+                {
+                    var content = file.ReadText();
+                    GitUtility.CheckMergeConflictMarker(content, file.FilePath);
+                    return ExtractYamlHeader.Extract(content);
+                })).Value;
+
             private string GetKeyFromFile(Document file)
```

Comment:
>GetKeyFromFile [](start = 27, length = 14)

do we need consider the path Comparison case sensitive or not?

Reply:
I think it should be case-sensitive here since we just want to identify if the file has been parsed